### ISSUE TITLE
Double base url appended

### ISSUE THREAD

**Issue by riksnelders** [JustinBeckwith/retry-axios#4]
Using /v1/api via:
```js
window.axios.defaults.baseURL = '/v1/api/'
```

On error results in 
 '/v1/api/v1/api'  call

**Comment by JustinBeckwith** [comment]
Yar.  Can you share a full code sample please?

**Comment by riksnelders** [comment]
```
window.axios = require('axios');
window.axios.defaults.baseURL = '/v1/api/'
window.axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
const rax = require('retry-axios');
const interceptorId = rax.attach(axios);

```

**Comment by riksnelders** [comment]
Fixed for now with using base url attracted from js

**Comment by cherouvim** [comment]
Confirming this bug.

- original call: /cms/en/api/page/about returns 503 (Service Unavailable)
- axios retries to: /cms/en/cms/en/api/page/about

**Comment by riksnelders** [comment]
This fixed yet? :)?

**Comment by klaky** [comment]
Have same issue

**Comment by Spyna** [comment]
Facing the same issue, and is not related to `retry-axios` but to `axios`. 
If you look at [`dispatchRequest.js`](https://github.com/axios/axios/blob/master/lib/core/dispatchRequest.js) in the axios source code you'll see near line 25.
```JS
  // Support baseURL config
  if (config.baseURL && !isAbsoluteURL(config.url)) {
    config.url = combineURLs(config.baseURL, config.url);
  }
```

If you don't use an absolute URL as `baseURL`, the `config.url` will be always combined with the `baseURL`. 

This means that..
if you do `window.axios.defaults.baseURL = '/v1/api/'`, the first time you execute a request, such as `axios.get('/user')` the actual `url` will be `[config.baseURL]` + `[config.url]` = ` '/v1/api/` + `user'` = ` '/v1/api/user'`. 

When that request is retried the code of *dispatchRequest.js* will be repeated, and now the value of `url` is already "prepended" with the baseURL. So the `config.url` will end up with  `[baseURL]` + `[url]` = ` '/v1/api/` + `/vi/api/user'` = ` '/v1/api/v1/api/user'`. 

**Solutions**

* Use an **absolute URL** as the `baseURL`
* Use a request interceptor to purge the `config.URL` if the request is being retried

**Comment by avindra** [comment]
I agree that this is more of an Axios issue. How do they manage to gaff URLS this badly? This is why I always recommend my teams to use basic functionality like `fetch`.

**Comment by avindra** [comment]
@riksnelders , others: I noticed in the [code posted in the OP](https://github.com/JustinBeckwith/retry-axios/issues/4#issue-311509169) has a relative URL, not an absolute URL.

Axios is making a lot of assumptions in their code that the URL MUST be absolute. An easy fix may be to set it to `location.origin + '/v1/your/api/base/here'`.

After making this change in our app, I noticed this bug went away. 

-----

Aside, this is actually a core axios expectation.

This is actually spelled out in the documentation, but its extremely easy to overlook (its in the comment above the `baseUrl` config in the code:

https://github.com/axios/axios#request-config

I think if enough heads work on the problem, they can clean it up and have `baseUrl` ALSO support relative paths, as that expectation works cleanly with just core axios. The sloppy handling of config values manifests an issue when trying to add the retry functionality.
